File: ucln.pas
program ucln;
uses crt;
//Bai lam cua : Ngo Duy Anh
var M,N:integer;
begin  
    clrscr;
    write('Nhap M, N = ');
    readln(M,N);
    while M <> N do
        if M > N then M:= M-N else N:= N-M;
    writeln('UCLN = ', M);
    readln
end.

File: ucln.cpp
#include <iostream>
using namespace std;

// Bai lam cua Ngo Duy Anh
long long M,N;

int main() {
    cout<<"M= ";    cin>>M;
    cout<<"N= ";    cin>>N;
    while(M !=N)    if (M>N) M=M-N; else N=N-M;
    cout << "UCLN =" << M << endl;
    return 0;
}